registry/single_master.go

```diff
@@ -0,0 +1,44 @@
+package registry
+
+import "github.com/mailgun/go-etcd/etcd"
+
+const (
+	masterNodeID = "master"
+)
+
+/*
+SingleMasterRegistry is an implementation of Registry that uses a single master
+instance of an application to handle requests. Internally, this registry uses
+a GroupMasterRegistry with a single, constant group ID.
+*/
+type SingleMasterRegistry struct {
```

Comment:
I would call this `LeaderFollower` registry and implement compare and swap here instead